### DISCUSSION TITLE
Adding new throttle blocks outside of config file

### DISCUSSION THREAD

**Discussion by aishbuilds** [kickstarter/rack-attack#661] [Questions (Q&A)]
Hi,

I want to dynamically throttle specific users in runtime, from a custom dashboard.
In this case, I cannot define the throttle blocks in the `rack-attack.rb` config file.

When I define the throttle list from the controller, it does not seem to work.
I am unable to find a workaround for this, any idea on how I can get this working?

Thanks!

**Comment by grzuy** [comment]
The only thing i can think of right now that might help with your use case would be storing/keeping some value in the `cookies` or `session` hashes that differentiate users.

Then in your `config/initializers/rack_attack.rb` you can have something like

```ruby
Rack::Attack.throttle("dashboard", limit: ..., period: ...) do |request|
  request.cookies[:user_id]
end
```

for example.

**Comment by grzuy** [comment]
I hope the above helps :-)
If not, let me know and we reopen.

**Comment by alexsmartens** [comment]
@grzuy, any way you've figured this one out?
